package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -288,20 +288,27 @@ def _write_atoms(self, atoms):
         indices = atoms.indices + 1
         types = atoms.types.astype(np.int32)
 
+        try:
+            moltags = data['molecule_tag']
+        except KeyError:
+            moltags = [0] * len(atoms)
+
```

Comment:
```suggestion
        moltags = data.get("molecule_tag", np.zeros(len(atoms), dtype=int))
```

Allocating a `numpy` array should be considerably faster. `get` allows to avoid the try-except logic.

```
$ N = 10000
$ %timeit [0] * N
23.5 µs ± 359 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

$ %timeit np.zeros(N, dtype=int)
2.85 µs ± 231 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
Thanks for this suggestion! This is clearly a better way.